ams/CleanupCommand.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            var account = await GetMediaAccountAsync(_options.AccountName, cancellationToken);
+            _logger.LogInformation("Begin cleaning up on account: {name}", account.Data.Name);
+
+            if (_options.IsCleanUpAccount)
+            {
+                Console.Write($"Do you want to delete the account '{account.Data.Name}'? (y/n): ");
+                string? userResponse = Console.ReadLine();
+
+                if (!(userResponse?.ToLower() == "y"))
+                {
+                    Console.WriteLine("Account cleanup canceled by user.");
+                    return;
+                }
+            }
+
+            Dictionary<string, bool> stats = new Dictionary<string, bool>();
+            var totalAssets = await QueryMetricAsync(
+                account.Id.ToString(),
+                "AssetCount",
+                cancellationToken: cancellationToken);
+
+            _logger.LogInformation("The total asset count of the media account is {count}.", totalAssets);
+            AsyncPageable<MediaAssetResource> assets;
+           
+            //clean up asset
+            var resourceFilter = _options.IsCleanUpAccount? null: GetAssetResourceFilter(_options.ResourceFilter, null, null);
+
+            var orderBy = "properties/created";
+            assets = account.GetMediaAssets()
+                .GetAllAsync(resourceFilter, orderby: orderBy, cancellationToken: cancellationToken);
+            List<MediaAssetResource>? assetList = await assets.ToListAsync(cancellationToken);
+
+            foreach (var asset in assetList)
+            {
+                var result = await CleanUpAssetAsync(_options.IsCleanUpAccount||_options.IsForceCleanUpAsset,account, asset, cancellationToken);
+                stats.Add(asset.Data.Name, result);
+            }
+            WriteSummary(stats, false);
+
+            if (_options.IsCleanUpAccount)
+            {
+                var endpoints = account.GetStreamingEndpoints();
+                var liveevents = account.GetMediaLiveEvents();
+                var policies = account.GetContentKeyPolicies();
+
+                if (endpoints != null)
+                {
+                    foreach (var streamingEndpoint in endpoints)
+                    {
+                        await streamingEndpoint.DeleteAsync(WaitUntil.Completed);
+                    }
+                }
+                if (policies != null)
+                {
+                    foreach (var contentKeyPolicy in policies)
+                    {
+                        await contentKeyPolicy.DeleteAsync(WaitUntil.Completed);
+                    }
+                }
+                if (liveevents != null)
+                {
+                    foreach (var liveEvent in liveevents)
+                    {
+                        await liveEvent.DeleteAsync(WaitUntil.Completed);
+                    }
+                }
+                _logger.LogDebug("account {account} is deleted.", account.Data.Name);
```

Comment:
Nit:  This log can be reported after CleanupAccountAsync( ) is called, or just move  the lines between 74 and 99 into function CleanupAccountAsync( ).